src/Compilers/CSharp/Test/Semantic/SourceGeneration/GeneratorDriverTests.cs

```
@@ -1783,5 +1783,163 @@ class C { }
+            compilation.VerifyDiagnostics();
+
+            Assert.Single(compilation.SyntaxTrees);
+
+            List<ParseOptions> parseOptionsCalledFor = new List<ParseOptions>();
+
+            var generator = new IncrementalGeneratorWrapper(new PipelineCallbackGenerator(ctx =>
+            {
+                ctx.RegisterSourceOutput(ctx.ParseOptionsProvider, (spc, p) => { parseOptionsCalledFor.Add(p); });
+            }));
+
+            // run the generator once, and check it was passed the parse options
+            GeneratorDriver driver = CSharpGeneratorDriver.Create(new ISourceGenerator[] { generator }, parseOptions: parseOptions);
+            driver = driver.RunGenerators(compilation);
+            Assert.Equal(1, parseOptionsCalledFor.Count);
+            Assert.Equal(parseOptions, parseOptionsCalledFor[0]);
+
+            // clear the results, and re-run
+            parseOptionsCalledFor.Clear();
+            driver = driver.RunGenerators(compilation);
+            Assert.Empty(parseOptionsCalledFor);
+
+            // now update the parse options
+            parseOptionsCalledFor.Clear();
+            var newParseOptions = parseOptions.WithDocumentationMode(DocumentationMode.Diagnose);
+            driver = driver.WithUpdatedParseOptions(newParseOptions);
+
+            // check we ran
+            driver = driver.RunGenerators(compilation);
+            Assert.Equal(1, parseOptionsCalledFor.Count);
+            Assert.Equal(newParseOptions, parseOptionsCalledFor[0]);
+
+            // clear the results, and re-run
+            parseOptionsCalledFor.Clear();
+            driver = driver.RunGenerators(compilation);
+            Assert.Empty(parseOptionsCalledFor);
+
+            // update the parse options to a null value
+            driver = driver.WithUpdatedParseOptions(null!);
+
+            // check the driver ignored the null update
```

>check the driver ignored the null update [](http://example.com/codeflow?start=15&length=40)

If we're expecting a non-null parse options, should we throw on `null` rather than silently ignoring the update? Same question for config options.

---

Hmm, yeah I agree actually we should probably just throw.